rep-2008.rst

```diff
@@ -0,0 +1,778 @@
+  
+
+* Enforcing specific hardware solutions on maintainers
+
+  * No maintainer is required to target any specific hardware solution by any of the guidelines in this REP.
+  * Instead, maintainers are free to choose among any of the supported technologies and hardware solutions.
+
+
+Architecture pillars
+====================
+
+:: 
+
+       ROS 2 stack                    Hardware Acceleration Architecture @ ROS 2 stack
+
+    +-------------+             +---------------------------+
+    |             |             |   acceleration_examples   |
+    | user land   |  +-------------------+-----------+----------------+--------------------------------+
+    |             |  |       Drivers     |     Libraries              |           Cloud                |
+    +-------------+  +--------------+----+---------+-----------------------------------+---------------+
+    |             |  |   ament_sol1 | ament_soln   |                  |  accel_fw_sol1 | accel_fw_soln |
+    |             |  +--------------+--------------+------------------+----------------+---------------+
+    |  tooling    |  |     ament_acceleration   | colcon_acceleration |       acceleration_firmware    |
+    |             |  +---------------------------------------------------------------------------------+
+    |             |  |      build system        |   meta build        |            firmware            |
+    +-------------+  +--------------------------+---------------------+--------------------------------+
+    |     rcl     |             ^                      ^                            ^
+    +-------------+             |                      |                            |
+    |     rmw     |             |                      |                            |
+    +-------------+             |                      |                            |
+    | rmw_adapter |          Pillar I              Pillar II                    Pillar III
+    +-------------+
+    
+
+
+.. _pillarI:
+
+Pillar I - Extensions to ament
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The first pillar represents extensions of the ``ament`` ROS 2 build system. These CMake extensions help achieve the objective of simplifying the creation of acceleration kernels. By providing an experience and a syntax similar to other ROS 2 libraries targeting CPUs, maintainers will be able to integrate kernels into their packages easily. The ``ament_acceleration`` ROS 2 package abstracts the build system extensions from technology-specific frameworks and software platforms. This allows to easily support hardware acceleration across FPGAs and GPUs while using the same syntax, simplifying the work of maintainers. The code listing below provides an example that instructs the ``CMakeLists.txt`` file of a ROS 2 package to build a ``vadd`` kernel with the corresponding sources and configuration:
+
+::
+
+    acceleration_kernel(
+      NAME vadd
+      FILE src/vadd.cpp
+      INCLUDE
+        include
+      TARGET kv260
```

Comment:
```suggestion
```

While consering https://discourse.ros.org/t/rep-2008-rfc-ros-2-hardware-acceleration-architecture-and-conventions/22026/8?u=vmayoral, I realized this `TARGET` argument is actually not needed for production, and a  simple leftover from the development process while debugging across different targets.

The embedded (or non-embedded) target itself can be identified by looking at the build arguments provided to `colcon` and/or matched with the firmware deployed and selected in the ROS 2 workspace. In a nutshell, it's not needed.